cylc/flow/simulation.py

```diff
@@ -29,14 +34,41 @@
 )
 from cylc.flow.wallclock import get_current_time_string
 
-from metomi.isodatetime.parsers import DurationParser
+from metomi.isodatetime.parsers import DurationParser, TimePointParser
 
 if TYPE_CHECKING:
     from queue import Queue
     from cylc.flow.cycling import PointBase
     from cylc.flow.task_proxy import TaskProxy
 
 
+@dataclass
+class ModeSettings:
+    """A store of state for simulation modes.
+
+    Used instead of modifying the runtime config.
+    """
+    simulated_run_length: float = 0.0
+    sim_task_fails: bool = False
+
+    def update(self, itask, broadcast_mgr):
+        """Update an instance of this object.
+        """
+        overrides = broadcast_mgr.get_broadcast(itask.tokens)
+        if overrides:
+            rtconfig = pdeepcopy(itask.tdef.rtconfig)
+            poverride(rtconfig, overrides, prepend=True)
+        else:
+            rtconfig = itask.tdef.rtconfig
+        self.simulated_run_length = (
+            set_simulated_run_len(rtconfig))
+        self.sim_task_fails = sim_task_failed(
+            rtconfig['simulation'],
+            itask.point,
+            itask.submit_num
+        )
```

Comment:
This could be in the `__init__()` method?

Reply:
No.

We might wish to broadcast a change to the outcome (probably only for skip mode really) between simulated job submissions, in which case we want to modify rather than create anew.

Reply:
The 2 instances where it's used both do
```
itask.mode_settings = ModeSettings()
itask.mode_settings.update(...)
```

Reply:
Fair enough, but I believe it would still have the same behaviour if you replaced it with
```diff
-if not itask.mode_settings:
-    itask.mode_settings = ModeSettings()
-itask.mode_settings.update(
+itask.mode_settings = ModeSettings(
     itask, self.task_events_mgr.broadcast_mgr)
```